exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +60,482 @@ type BTCMarkets struct {
+}
 
-	req := WithdrawRequestAUD{
-		AccountName:   accountName,
-		AccountNumber: accountNumber,
-		BankName:      bankName,
-		BSBNumber:     bsbNumber,
-		Amount:        newAmount,
-		Currency:      "AUD",
-	}
+// CreateNewReport creates a new report
+func (b *BTCMarkets) CreateNewReport(reportType, format string) (ReportData, error) {
+	var resp ReportData
+	req := make(map[string]interface{})
+	req["type"] = reportType
+	req["format"] = format
+	return resp, b.SendAuthenticatedRequest(http.MethodPost, btcMarketsReports, req, &resp, nil)
+}
 
-	resp := Response{}
-	err := b.SendAuthenticatedRequest(http.MethodPost, btcMarketsWithdrawAud,
-		req,
-		&resp)
-	if err != nil {
-		return "", err
+// GetReport creates a new report
+func (b *BTCMarkets) GetReport(reportID string) (ReportData, error) {
+	var resp ReportData
+	path := btcMarketsReports + "/" + reportID
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, path, nil, &resp, nil)
+}
+
+// RequestWithdraw requests withdrawals
+func (b *BTCMarkets) RequestWithdraw(assetName string, amount float64,
+	toAddress, accountName, accountNumber, bsbNumber, bankName string) (TransferData, error) {
+	var resp TransferData
+	req := make(map[string]interface{})
+	req["assetName"] = assetName
+	req["amount"] = strconv.FormatFloat(amount, 'f', -1, 64)
+	if assetName != "AUD" {
+		req["toAddress"] = toAddress
+	} else {
+		if accountName != "" {
+			req["accountName"] = accountName
+		}
+		if accountNumber != "" {
+			req["accountNumber"] = accountNumber
+		}
+		if bsbNumber != "" {
+			req["bsbNumber"] = bsbNumber
+		}
+		if bankName != "" {
+			req["bankName"] = bankName
+		}
 	}
+	return resp, b.SendAuthenticatedRequest(http.MethodPost, btcMarketsWithdrawals, req, &resp, nil)
+}
 
-	if !resp.Success {
-		return "", errors.New(resp.ErrorMessage)
+// BatchPlaceCancelOrders places and cancels batch orders
+func (b *BTCMarkets) BatchPlaceCancelOrders(cancelOrders []cancelBatch, placeOrders []placeBatch) (BatchPlaceCancelResponse, error) {
+	var resp BatchPlaceCancelResponse
+	var orderRequests []interface{}
+	for x := range cancelOrders {
+		orderRequests = append(orderRequests, cancelOrders[x])
+	}
+	for y := range placeOrders {
+		orderRequests = append(orderRequests, placeOrders[y])
 	}
+	return resp, b.SendAuthenticatedRequest(http.MethodPost, btcMarketsBatchOrders, nil, &resp, orderRequests)
+}
 
-	return resp.Status, nil
+// GetBatchTrades gets batch trades
+// ids MUST be comma separated with no spaces
+func (b *BTCMarkets) GetBatchTrades(ids string) (BatchTradeResponse, error) {
+	var resp BatchTradeResponse
+	path := btcMarketsBatchOrders + "/" + ids
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, path, nil, &resp, nil)
+}
+
+// CancelBatchOrders cancels given ids
+// ids MUST be comma separated with no spaces
+func (b *BTCMarkets) CancelBatchOrders(ids string) (BatchCancelResponse, error) {
```

Comment:
This could also be a string array and then you just do a `strings.Join(strArray, ",")` which would ensure the formatting is correct and no need for the comment above